packages/client/wallets/aa/src/blockchain/wallets/FireblocksNCWallet.ts

```diff
@@ -111,7 +117,12 @@ export function getSmartAccountSignerFromFireblocks(
 ): SmartAccountSigner {
     return {
         getAddress: async () => {
-            return (await crossmintService.getAddress(walletId, 0, getAssetIdByBlockchain(chain))) as `0x${string}`;
+            let address = localStorage.getItem(NCW_ADDRESS_KEY);
+            if (!address) {
+                address = await crossmintService.getAddress(walletId, 0, getAssetIdByBlockchain(chain));
+                storeAddressToLocalStorage(address!);
+            }
+            return address as `0x${string}`;
```

Comment:
similarly, if we add `const localStorageRepository = new LocalStorageRepository()` on the line before `return {`, then we can:
```suggestion
            let address = localStorageRepository.ncwAddress;
            if (!address) {
                address = await crossmintService.getAddress(walletId, 0, getAssetIdByBlockchain(chain));
                localStorageRepository.ncwAddress = address;
            }
            return address as `0x${string}`;
```